### ISSUE TITLE
Incoming data: handle image URLs that 404

### ISSUE THREAD

**Issue by davewhiteland** [everypolitician/everypolitician#472]
# Problem

When reconciling against incoming twitter data, the twitter image is sometimes broken because the URL 404s. That "bad" URL ends up in the data even though at this point we know it's not a useful image.

![screen shot 2016-07-22 at 15 13 10](https://cloud.githubusercontent.com/assets/111320/17059870/229f0b7c-501f-11e6-811d-4e8868ec9d5c.png)

I _think_ this happens because people can readily change their profile image (not sure).
# Proposed Solution

Currently I don't think we notice the failure, but if we loaded that in the reconciliation tool with some JavaScript we could know when these are 404ing. But that presupposes it is being run through the reconciliation tool.

So maybe there's a case for testing health of any image URL that's added to the data?
# Acceptance Criteria

Image URLs that do not yield images (specifically 404 returned instead) at time of import do not get added to the data.
# Not Required

—
# Prerequisites

—
# Related Issues

—
# Due Date

—
# Design?

No.
# Text?

No.
# Bloggable?

Probably not.

**Comment by tmtmtmtm** [comment]
I'm not sure I agree with this suggestion. 

If we care about whether an image URL is 404ing, then just checking that at the point where we add it to the data isn't going to suffice (e.g. if the image disappears 3 months later). And surprisingly often an image will fail to load today, but magically reappear tomorrow.

Whether we _should_ care or not is even an interesting question. From a very strict POV, EP is simply reporting that a source is currently telling us that there's an image for this person at that URL.  Even if that image has vanished permanently, there are all sorts of useful reasons for us to still include the URL (maybe a consuming app has cached the image and can still display it based on a sha of the URL; or can fetch it from the wayback machine; etc).

**Comment by davewhiteland** [comment]
@tmtmtmtm agreed -- I wasn't sure about this suggestion either, so raised it speculatively.

I think pragmatically it _is_ OK to consider a 404 to _not_ be transient (after all, if it's really there 24 hours later, we'll pick it up again). But I agree there's a broader case for being unconcerned by the 404s but to me that's more because we don't know the cause of it. So until we do, we can assume the URL is potentially useful.

It's the nature of "the image problem" (namely, timely, volatile URLs) especially that we're going to get linkrotted URLs accumulating of course, but that's the nature of the web as an archive after all. Hence WayBack, yeah.

Another good counterargument is there is certainly some value in retaining poiticians' `website` URLs even though we're not checking for DNS health/404s there before including them. 

So, yup, happy for you to close this now.

**Comment by wfdd** [comment]
I remove broken images in the Colombian Senate scraper just so that the image cacher will use the Twitter photos instead in those cases - something to bear in mind.

**Comment by davewhiteland** [comment]
Yes @wfdd that is a good point. I think that's the pragmatic solution from EveryPolitician's point of view: if upstream wants to worry/handle 404s, that's the right place for it. So it's not really something the merge/reconcile process needs to get bogged down in.

**Comment by wfdd** [comment]
Well, it's a technical compromise - I'd have no particular reason to remove those URLs if it weren't for a limitation of the image cacher.  I could actually revert that change now that the thumbnails have been generated.  Anyway, my point was some other people might've done the same and it's generally a good idea to be cognisant of how the tools that we use can lead us to make the wrong (or right) choices.
